client/src/app/auth/components/register/register.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthService } from '../../services/auth.service';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  registerForm: FormGroup
+
+  formErrors: any = {
+    userName: '',
+    email: '',
+    password: '',
+  }
+
+  constructor(
+    private authService: AuthService
+  ) { }
+
+  ngOnInit(): void {
+    this.registerFormInit()
+  }
+
+  registerFormInit(): void {
+    this.registerForm = new FormGroup({
+      userName: new FormControl(null, [Validators.required, Validators.minLength(3)]),
+      email: new FormControl(null, [Validators.required, Validators.email]),
+      password: new FormControl(null, [Validators.required, Validators.pattern('')])
+    })
+  }
+
+  submitRegisterForm(): void {
+    if (this.registerForm.invalid) return
+    this.registerForm.disable()
```

Comment:
do not write code in same way. Please use semicolon and brackets, this ts code could be compiled in wrong JS file